src/mon/AuthMonitor.h

```diff
@@ -127,16 +130,28 @@ class AuthMonitor : public PaxosService {
     pending_auth.push_back(inc);
   }
 
-  /* validate mon caps ; don't care about caps for other services as
+  /* validate mon/osd/mds caps ; don't care about caps for other services as
    * we don't know how to validate them */
   bool valid_caps(const vector<string>& caps, ostream *out) {
     for (vector<string>::const_iterator p = caps.begin();
          p != caps.end(); p += 2) {
-      if (!p->empty() && *p != "mon")
-        continue;
-      MonCap tmp;
-      if (!tmp.parse(*(p+1), out))
-        return false;
+      if (!p->empty() && *p == "mon") {
+        MonCap tmp;
+        if (!tmp.parse(*(p+1), out))
+          return false;
+      }
+
+      if (!p->empty() && *p == "osd") {
+        OSDCap ocap;
+        if (!ocap.parse(*(p+1), out))
+          return false;
+      }
+
+      if (!p->empty() && *p == "mds") {
+        MDSAuthCaps mdscap;
+        if (!mdscap.parse(g_ceph_context, *(p+1), out))
+          return false;
+      }
```

Comment:
This construction appears to validate the caps (by falling through all the test cases) if it doesn't match any of the known cap users. That seems bad?

Reply:
Here it only try to validate the caps when it's type is either "mon" or "osd" or "mds" and won't validate any other unknown types. 
Do you mean that it's bad for unknown types?

Reply:
Yeah, I don't think we want to accept capabilities that don't match a known type. Otherwise somebody could typo and end up with caps on the "min" and none on the mon, or similar.